### DISCUSSION TITLE
C++ Support via Node-API

### DISCUSSION THREAD

**Discussion by thompsonnoahe** [elgatosf/elgato#145] [Help] [answered]
Hello,

I'm currently maintaining a new plugin called GPU Utilization that leverages [this C++ SDK](https://github.com/fredemmott/StreamDeck-CPPSDK).

It'd be a significant effort to port everything over to Node, however the transition would be worthwhile if there is better support for this version and if it's the recommended route. Further, it'd be great if we could have a sample plugin demonstrating how to achieve this.

Thank you!

**Comment by ChekTek** [comment, marked as answer]
A great request! We use the C++ NAPI for some plugins internally, a sample demonstrating this would be a great idea. I will add this to our road map.

**Reply by thompsonnoahe** [reply]
That sounds great! Thank you!
